dotCMS/src/main/java/com/dotcms/rest/BundlePublisherResource.java

```diff
@@ -45,140 +54,178 @@
-		                Logger.error(this, "error close session",e);
-		            }
+			return new WebResource.InitBuilder().
+					rejectWhenNoUser(false). // it would be a soft validation so not reject
+					requestAndResponse(
+					new HttpHeaderHandlerHttpServletRequestWrapper(request,
+							CollectionsUtils.map(
+									"Authorization", (name, value) -> // if the authorization is set, uses it, otherwise try with the secret (could be a jwt)
+											UtilMethods.isSet(value) ? value : JsonWebTokenAuthCredentialProcessor.BEARER + authTokenDigest
+							)), response).init();
+		}catch (Exception e) {
+    		return null;
+		}
+
+	}
+
+
+	@WrapInTransaction
+	private Bundle publishBundle(final String fileNameSent,
+								  final String groupId,
+								  final String endpointId,
+								  final String bundleNameSent,
+								  final boolean forcePush,
+								  final HttpServletRequest request,
+								  final String remoteIP,
+								  final PublishingEndPoint sendingEndPointByAddress) throws Exception {
+
+    	final String fileName = UtilMethods.isSet(fileNameSent) ? fileNameSent : generatedBundleFileName();
+		final String bundleName =  UtilMethods.isSet(bundleNameSent) ? bundleNameSent : fileName;
+
+		Bundle bundle = null;
+
+		try (InputStream bundleStream = request.getInputStream()) {
+
+			final String bundlePath         = ConfigUtils.getBundlePath()+ File.separator + MY_TEMP;
+			final String bundleFolder       = fileName.substring(0, fileName.indexOf(".tar.gz"));
+			final String sendingEndPoint = sendingEndPointByAddress != null ? sendingEndPointByAddress.getId() : remoteIP;
+			final PublishAuditStatus status = PublishAuditAPI.getInstance().updateAuditTable(
+					sendingEndPoint, sendingEndPoint, bundleFolder, true);
+
+			if(bundleName.trim().length() > 0) {
+				// save bundle if it doesn't exists
+				bundle = APILocator.getBundleAPI().getBundleById(bundleFolder);
+				if (bundle == null || bundle.getId() == null) {
+
+					bundle = new Bundle();
+					bundle.setId(bundleFolder);
+					bundle.setName(bundleName);
+					bundle.setPublishDate(Calendar.getInstance().getTime());
+					bundle.setOwner(APILocator.getUserAPI().getSystemUser().getUserId());
+					bundle.setForcePush(forcePush);
+					APILocator.getBundleAPI().saveBundle(bundle);
 				}
-    		}
-    	} catch (IOException e) {
-    		Logger.error(PublisherQueueJob.class,e.getMessage(),e);
-    	}
+			}
+
+			//Write file on FS
+			FileUtil.writeToFile(bundleStream, bundlePath + fileName);
+
+			//Start thread
 
-		return Response.status(HttpStatus.SC_INTERNAL_SERVER_ERROR).build();
+			if(!status.getStatus().equals(Status.PUBLISHING_BUNDLE)) {
+				DotConcurrentFactory.getInstance()
+						.getSubmitter()
+						.submit(new PublishThread(fileName, groupId, endpointId, status));
+			}
+
+			return bundle;
+		} catch (Exception e) {
+
+			Logger.error(PublisherQueueJob.class, "Error caused by remote call of: "+remoteIP);
```

Comment:
Include endpointId and bundleNameSent variables in error logging. Use the same error message for both the logging and the exception's message

Reply:
https://github.com/dotCMS/core/pull/18555/commits/1aa6ff79d0ee2e5d6131b4a519f93d43a955a497#diff-4beff38a2c10d3a1c1843475075d0893R198